app/templates/admin/userManagement.html

```diff
@@ -114,36 +121,124 @@ <h3 class="accordion-header" id="headingTwo">
-      <div id="collapseThree" class="accordion-collapse collapse {{show}}" aria-labelledby="headingThree" data-bs-parent="#adminManagement">
-        <div class="accordion-body">
-          <table>
-            <tr>
-              <th>Program Name</th>
-              <th>Contact Email</th>
-              <th>Contact Name</th>
-              <th>Location</th>
-            </tr>
-            {% for program in programs %}
-            <form action="/admin/updateProgramInfo/{{program.id}}" method="POST">
-              <tr class="p-1">
-                <td><input class="form-control" type="text" name="programName" value ="{{program.programName}}" required></td>
-                <td><input class="form-control" type="text" name="contactEmail" value="{{program.contactEmail or ""}}"></td>
-                <td><input class="form-control" type="text" name="contactName" value="{{program.contactName or ""}}"></td>
-                <td><input class="form-control" type="text" name="location" value="{{program.defaultLocation or ""}}"></td>
-                <td><button type="submit" class="btn btn-success w-101" role ="button">Update</button> </td>
+      <div id="collapseThree" class="accordion-collapse collapse {{ show }}" aria-labelledby="headingThree" data-bs-parent="#adminManagement">
+        <div class="accordion-body ml-5">
+          <div class="d-flex justify-content-center">
+           <table>
+              <tr>
+                 <th class="programHeader">Program</th>
+              </tr>
+              
+              {% for program in programs %}
+              <tr>
+                 <td>
+                  <input class="form-label borderless" 
+                    name="programName" 
+                    type="text" readonly 
+                    value="{{ program.programName }}"></td>
+                 <td>
+                    <button class="btn btn-success test" 
+                    role="button" id="programManagementButton{{ loop.index }}" 
+                    data-bs-toggle="modal" 
+                    data-bs-target="#adminProgramManagement"
+                    onclick="programModal('{{ program.programName }}', '{{ program.programDescription }}', '{{ program.partner }}', '{{ program.contactEmail }}', '{{ program.contactName }}', '{{ program.defaultLocation }}', '{{ program.id }}', '{{ program.instagramUrl }}','{{ program.facebookUrl }}', '{{ program.bereaUrl }}')">Update Program Information</button>
+                    <script 
+                      type="module"> import { populateModal } from "/static/js/userManagement.js";
+                      window.programModal = function(programName, programDescription, partner, contactEmail, contactName, location, programId, instagramUrl, facebookUrl, bereaUrl){
+                      populateModal(programName, programDescription, partner, contactEmail, contactName, location, programId, instagramUrl, facebookUrl, bereaUrl)
+                      }
```

Comment:
If we store the program info as a data element `data-programInfo="{{ programData }}"`, then we can add an onclick to all the buttons from javascript that grabs this data from whichever button is the `event.target`.  The snag is getting a JSON representation of the dictionary for the Program model object. I'm not sure how to do that. `model_to_dict` or some other method.

Reply:
We utilized JSON representation from the JS to directly populate the modal fields.